packages/flame/lib/src/effects/opacity_effect.dart

```diff
@@ -55,36 +61,43 @@ class OpacityEffect extends ComponentEffect<HasPaint> {
         );
 
   @override
-  void initialize(HasPaint component) {
-    super.initialize(component);
+  Future<void> onLoad() async {
+    super.onLoad();
     peakTime = duration;
 
-    _original = component.getPaint(paintId).color;
-    _final = _original.withOpacity(opacity);
+    _original = affectedParent.getPaint(paintId).color;
+    _peak = _original.withOpacity(opacity);
 
     _difference = _original.opacity - opacity;
+    setPeakTimeFromDuration(duration);
   }
 
   @override
-  void setComponentToEndState() {
-    component?.setColor(
-      _final,
+  void setComponentToPeakState() {
+    affectedParent.setColor(
+      _peak,
       paintId: paintId,
     );
   }
 
   @override
   void setComponentToOriginalState() {
-    component?.setColor(
+    affectedParent.setColor(
       _original,
       paintId: paintId,
     );
   }
 
   @override
   void update(double dt) {
+    if (isPaused) {
+      return;
+    }
     super.update(dt);
-    component?.setOpacity(
+    if (hasCompleted()) {
```

Comment:
Comparing the changes on this Effect and the changes on the Color effect, seems that we are repeating quite an amount of code on the effects, we should consider thinking on a way to share more logic on our effects library (Also non blocker for this PR)

Reply:
should we add an issue just to keep track of the followups? I fear some good comments might get lost when this is merged

Reply:
Opened #945